pkg/tfbridge/provider.go

```diff
@@ -324,10 +325,19 @@ func (p *Provider) CheckConfig(ctx context.Context, req *pulumirpc.CheckRequest)
 		}
 		if p.info.PreConfigureCallbackWithLogger != nil {
 			// NOTE: the user code may modify news in-place.
-			validationErrors := p.info.PreConfigureCallbackWithLogger(ctx, p.host, news, config)
-			if validationErrors != nil {
-				return nil, validationErrors
+			var logErr multierror.Error
+			ctx := ctxWithHostLogger(ctx, p.host, resource.URN(req.GetUrn()), &logErr)
+			err := p.info.PreConfigureCallbackWithLogger(ctx, p.host, news, config)
+			// If the only error is from the check itself, we let that through as-is,
+			// because it might be a grpc aware error.
+			if err != nil && len(logErr.Errors) == 0 {
```

Comment:
I'm not sure I understand error handling here, are we trying to promote accumulated errors to an error result from Check? What's an example where we need to do this?